File: javascript/src/Behavioral/Command/DvdName.js
(function() {
  define([], function() {
    var DvdName;
    return DvdName = (function() {
      DvdName.prototype.titleName = null;

      function DvdName(titleName) {
        this.titleName = titleName;
      }

      DvdName.prototype.setTitleName = function(titleNameIn) {
        return this.titleName = titleNameIn;
      };

      DvdName.prototype.getTitleName = function() {
        return this.titleName;
      };

      DvdName.prototype.setNameStarsOn = function() {
        return this.setTitleName(this.getTitleName().replace(' ', '*'));
      };

      DvdName.prototype.setNameStarsOff = function() {
        return this.setTitleName(this.getTitleName().replace('*', ' '));
      };

      DvdName.prototype.toString = function() {
        return "DVD: " + (this.getTitleName());
      };

      return DvdName;

    })();
  });

}).call(this);

File: ruby/src/Behavioral/Command/DvdName.rb
#//DvdName  the Receiver


class DvdName

	def initialize(titleName)
 		setTitleName(titleName)
	end

	def setTitleName(titleNameIn)
 		@titleName = titleNameIn
	end

	def getTitleName()
 		@titleName
	end

	def setNameStarsOn()
 		setTitleName(getTitleName().gsub(' ','*'))
	end

	def setNameStarsOff()
 		setTitleName(getTitleName().gsub('*',' '))
	end

	def to_s()
		return "DVD: #{getTitleName()}"
	end
end
